### ISSUE TITLE
Rename "Account X" / Confirm creating new / Delete

### ISSUE THREAD

**Issue by ivnnv** [leather-io/extension#1703]
Hi,

Just installed and used the extension for the first time for buying my very first NTF ever.

There a couple of things that I consider veeery "NiceToHave" at this point which are:

1) Ability to rename an account: ex: `Account 1` ---> `My Stacks Main Account`.

2) As been unable to `1)` I tried to create another account wondering / expecting if there will be a kind of "form" to creating it with a custom name, but this wasnt the case... a new account was created **straight** named `Account 2` 😓 
So now I have two accounts with two generic names that I hate haha.

3) Can an account be ever "Deleted"?
This is not that important if `1)` can be achieved, one could rename it as `Testing Mainnet Account` or something like that 😬 

Other than that, congrats for the project, keep up the good work 👍🏽

**Comment by 314159265359879** [comment]
>     1. Ability to rename an account: ex: `Account 1` ---> `My Stacks Main Account`.

If you register a BNS name on an account, then "Account X" will automatically be replaced by the name you registered. You can register a .btc name on btc.us.

>     3\. Can an account be ever "Deleted"?
>         This is not that important if `1)` can be achieved, one could rename it as `Testing Mainnet Account` or something like that 😬

It is currently not possible to remove an account, I think technically once you initialize a key it will always be there but I can imagine a user wanting to get rid of a specific key, perhaps in that case a _hiding option_ would suffice (it can still be accessed but will be hidden until you click _see hidden addresses_ or something like that).

**Comment by markmhendrickson** [comment]
Thanks for this set of suggestions.

Related: https://github.com/blockstack/stacks-wallet-web/issues/696

**Comment by markmhendrickson** [comment]
The ability to rename privately without having to register via BNS has also been requested here: https://www.reddit.com/r/stacks/comments/pqk09x/hiro_web_wallet_account_names/

**Comment by ivnnv** [comment]
I can confirm purchasing a btc.us domain shows the linked name to the account in the wallet, which is not ideal but its something! Plus I have my own btc domain for the next 5 years when we reach the mooonn! haha 🚀 
Thanks for the workaround suggestion @markmhx

**Comment by fbwoolf** [comment]
@markmhx this has been requested again (just for ref): https://github.com/blockstack/stacks-wallet-web/issues/1832

**Comment by jjenzz** [comment]
I purchased a BNS name but my wallet still reads "Account 1" even though the account lists the `name-register` transaction. I'm using Brave Browser, Hiro Wallet v2.19.4. 

Is there something else I need to do to replace it?

**Comment by fbwoolf** [comment]
> I purchased a BNS name but my wallet still reads "Account 1" even though the account lists the `name-register` transaction. I'm using Brave Browser, Hiro Wallet v2.19.4.
> 
> Is there something else I need to do to replace it?

Can you paste in your address so I can look into it?

**Comment by jjenzz** [comment]
@fbwoolf sure, thanks 🙂 `SP1W5YWMDBCBA4SCGTQB0F1WJ7VJKDEF6F4QPMW4P`

**Comment by fbwoolf** [comment]
> @fbwoolf sure, thanks 🙂 `SP1W5YWMDBCBA4SCGTQB0F1WJ7VJKDEF6F4QPMW4P`

The api does show you own it, so not sure why it isn't showing up. I'd almost say it might be a caching issue? I wonder if you just sign out and try logging in again to the wallet?

https://stacks-node-api.mainnet.stacks.co/v1/addresses/stacks/SP1W5YWMDBCBA4SCGTQB0F1WJ7VJKDEF6F4QPMW4P

**Comment by jjenzz** [comment]
> I wonder if you just sign out and try logging in again to the wallet?

I had already tried that before reporting here, sorry, should have mentioned that 🙈

**Comment by fbwoolf** [comment]
> > I wonder if you just sign out and try logging in again to the wallet?
> 
> I had already tried that before reporting here, sorry, should have mentioned that 🙈

I do see mine, so maybe a fresh install? Make sure to save your secret key! If that doesn't work, I would open a separate issue bc I think this one is more requesting to be able to change the name in the wallet directly.

<img width="351" alt="Screen Shot 2021-10-15 at 7 44 54 PM" src="https://user-images.githubusercontent.com/6493321/137567095-4dae20a7-df2f-48cd-9657-5331fc667d84.png">

**Comment by jjenzz** [comment]
> If that doesn't work, I would open a separate issue

No need, a fresh install solved it. Thank you 💛

**Comment by markmhendrickson** [comment]
> I'd almost say it might be a caching issue? 

@fbwoolf think it may be worth opening a separate issue to deal with busting the cache here more reliably for names?

**Comment by fbwoolf** [comment]
> > I'd almost say it might be a caching issue?
> 
> @fbwoolf think it may be worth opening a separate issue to deal with busting the cache here more reliably for names?

Yep, seems like a good idea. I can make a separate issue and close this one.

**Comment by fbwoolf** [comment]
Just realized this was a different issue so reopened, my bad.

**Comment by Rapha-btc** [comment]
I like the idea of being able to do some housekeeping, especially after having airdropped a few .BTC domains to my Acid Heart material frens. Werner pointed out this thread to me. Thank you frens. This will come handy:

Have an option to at least remove or hide an account from the UI on the wallet.

Ps: would have been great to be able to send an account to cant-be-evil.stx thou

**Comment by 314159265359879** [comment]
I would like to keep this open. It is still a popular feature. We would like to add it some day.
